### DISCUSSION TITLE
[sale] HCPBridge2 Board Set

### DISCUSSION THREAD

**Discussion by Gifford47** [Gifford47/HCPBridgeMqtt#83] [General]
# 🚪 HCPBridge & HCPMini — Development Boards for Hörmann Supramatic E/P & Promatic Series 4

Hi everyone 👋

I had a small batch of two board variants made for my own experiments with the [HCPBridge project](https://github.com/Gifford47/HCPBridgeMqtt) and have a few left over. I'm passing the remainder on to other tinkerers at cost.

> [!NOTE]
> These are **bare development boards for private experimentation**.
> Handed over as-is between hobbyists: no warranty, no returns but boards are fully tested.

Both boards can run the firmware from this [git repo](https://github.com/Gifford47/HCPBridgeMqtt/releases).

---

## 📊 The two variants

|                      | 🟦 **HCPBridge** Rev. 2.2                                               | 🟩 **HCPMini** 🆕                                                              |
| -------------------- | ----------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
| **Board size**       | 90 mm diameter                                                          | 56 × 58 mm                                                                     |
| **Placement**        | Intended for use outside the motor housing                              | Small enough for inside the motor housing, works outside too                    |
| **MCU**    | ESP32 (included)                                                         | ESP32 (included)                                                             |
| **RS485**            | Interface on board                                                      | Interface on board                                                             |
| **Expansion**        | Headers for temp / humidity / gas / distance / motion sensors           | Focused on motor control (optional: DHT22, DS18X20, BME280, distance, motion)   |
| **Typical use**      | Experimenting with a larger sensor setup around the door                | Minimal, discreet build                                                        |
| **Cost share**       | 42 € + shipping                                                         | 42 € + shipping                                                                |

---

## 🟦 HCPBridge Rev. 2.2

The larger of the two, laid out for experimenting with additional sensors around the door.

|                                                                                                       |
| :---------------------------------------------------------------------------------------------------: |
| ![HCPBridge Rev. 2.2](https://github.com/user-attachments/assets/79d81670-72e2-412a-94fa-56cd5894ea36) |
| **HCPBridge Rev. 2.2**                                                                                 |

| ![Enclosure with lid](https://github.com/user-attachments/assets/05e493e4-725e-4c72-b827-eef17e6ea7ef) | ![Enclosure with flipped lid](https://github.com/user-attachments/assets/5716b8d7-a424-4648-8388-4e203d11756b) | ![Enclosure complete](https://github.com/user-attachments/assets/c1367371-b95d-4019-91d4-282e06318e57) |
| :----------------------------------------------------------------------------------------------------: | :------------------------------------------------------------------------------------------------------------: | :----------------------------------------------------------------------------------------------------: |
| Example enclosure (self-printed, not included)                                                          | Lid flipped                                                                                                    | Assembled example                                                                                       |

**Board details**

- Designed against the Supramatic E/P Series 4 and Promatic Series 4 bus
- ESP32
- RS485 interface on board
- 2× input signals with 5V → 3V voltage divider
- 2× opto-isolated outputs (5V or 3V selectable)
- Layout intended for the outer compartment — see the [enclosure discussion](https://github.com/Gifford47/HCPBridgeMqtt/discussions/83#discussioncomment-10668439) for a printable case
- Optional sensor footprints: DS18X20 · BME280 · DHT22 · MQ4 · HC-SR04 · HC-SR501

---

## 🟩 HCPMini 🆕

Same idea, stripped down to 56 × 58 mm so it can disappear inside the motor housing.

| ![HCPMini front](https://github.com/user-attachments/assets/fb54bfec-da39-4bad-a57d-26d89656b234) | ![HCPMini installed](https://github.com/user-attachments/assets/2deca0f4-bcfe-442b-a74b-671d553e13e5) |
| :-----------------------------------------------------------------------------------------------: | :---------------------------------------------------------------------------------------------------: |
| HCPMini — top                                                                                     | HCPMini — example placement inside a motor                                                             |

**Board details**

- Designed against the Supramatic E/P Series 4 and Promatic Series 4 bus
- ESP32
- RS485 interface on board
- 56 × 58 mm, fits inside or outside the housing

---

## ❌ Not included

Bare board only. No pin headers, no sensors, no wiring, no enclosure.

> [!TIP]
> Headers and sensors aren't needed to talk to the motor; they only exist if you want to extend the board. All of it is standard maker stock from the usual sources.

---

## 📬 Interested?

Drop me a mail at **[Gifford47@protonmail.com](mailto:Gifford47@protonmail.com)** — copy the block below and fill in your details:

```
Amount of boards: 1
Product: HCPBridge         ← or HCPMini
First and last name: YOUR NAME
Postal code: 12345
City: MYCITY
Street + streetnumber: MYSTREET 11
Country: MYCOUNTRY
Insured parcel: yes|no  (higher shipping price)
```

Please only ask if you're comfortable working on your own hardware — I can't take responsibility for what happens on your garage door.

---

>[! Shipping notice]
>Important Shipping Notice:
For uninsured shipping, the customer assumes full responsibility for the shipment. I am not liable for lost or undelivered packages, and no refund will be issued in such cases.

**Comment by Gifford47** [comment]
Here's the enclosure:
[enclosure.zip](https://github.com/user-attachments/files/23996874/enclosure.zip)

**Comment by Gifford47** [comment]
Reserved for CAD

**Comment by skreuter** [comment]
@Gifford47 Thanks for this awesome board! Works like a charm 👍 
PS: I'm using esphome-hcpbridge on it

**Comment by toothman** [comment]
@Gifford47 Thank you from my side as well! All worked like a charm as well in the end. 
I'm also using esphome-hcpbridge wrapper btw :) 

One tiny question: I'd like to enable the i2c bus to configure an additional BME280 to send in some temperature data. On what pins on the ESP are SDA and SCL routed through? It would be great to add this to the docs :)

**Reply by 14yannick** [reply]
Hi i think you can use the pin definition from the config [file](https://github.com/Gifford47/HCPBridgeMqtt/blob/a3761a64eb3aaf9292952a013aa3aed79b763ed8/HCPBridgeESP32/src/configuration.h#L78)

**Reply by toothman** [reply]
Awesome - Thanks!

**Comment by Ten-Cars** [comment]
Hi @Gifford47 
This looks really cool. I just started recently with Home Automation (Home Assistant) and I'm not fully up to speed (yet). 

Can you help me with verifying if it would be compatible with my garage door system?
It is a Berner GA203 BS (https://www.berner-torantriebe.de/media/ga203-404_bs_de.pdf sorry only found a German manual). I can also provide pictures of the internal circuit board if that helps. I believe it uses HCP2, but I'm not 100% sure. 

How much would be shipping to a German address? 

Thanks a lot

**Comment by machine-bit** [comment]
Hi @Gifford47, looking forward to getting the boards. What does the case look like? Thanks

**Reply by Gifford47** [reply]
If I'll have some spare time, I'll offer the case. Sorry for the delay ...

**Reply by Gifford47** [reply]
The first enclosure for boards without sensors is ready. Look at the [first post](https://github.com/Gifford47/HCPBridgeMqtt/discussions/83#discussion-6917946).

**Comment by Gifford47** [comment]
Please use [discussion section](https://github.com/Gifford47/HCPBridgeMqtt/discussions) for all other questions besides from the ready to use set.

**Comment by Gifford47** [comment]
The first enclosure for boards without sensors is ready. Look at the [first post](https://github.com/Gifford47/HCPBridgeMqtt/discussions/83#discussion-6917946).

**Comment by BBNNL** [comment]
Is it possible to use the HCPBridge yaml with an Atom S3 Lite using an RS-485 converter?
![Atom S3 Lite +RS-485](https://github.com/user-attachments/assets/d68d431a-21d9-4751-a699-fcc88b2a5722)

**Reply by 14yannick** [reply]
I used this hardware for a couple of month and have a very reliable Setup without any issues.

**Reply by MKFY** [reply]
can it be powered direcly from hcp? according to the documentation there is a stepdown buck capable to convret from 36v onboard.

**Reply by apateluk** [reply]
Is this the AtomS3 Lite module ? I have been trying the same setup with the Atom Lite (ESP32 Pico) and can't seem to get the BUS scan to recognise it and also what pin settings have you changed for TX and RX ? Thanks.

[Edit] I have this now working with the M5 Atom Lite (Had the RX and TX the wrong way around. Had to modify the src to add support for the platform and also had to configure the pins as follows:

```
#define PIN_TXD 22
#define PIN_RXD 19

//Below may not need to be changed, but the pins originally overlapped with the TXD above.
#define I2C_SDA 25
#define I2C_SCL 21

```
Connect "B" to pin4 on the RJ12 connector
Connect "A" to pin 3 one the RJ12 connector
Pin 1&2: GND
Pin 5&6: +24V

**Reply by 14yannick** [reply]
Hi, yes I use the Atom S3. Swapping RX and TX is always a good first step in trial and error.
I use the [esphome](https://github.com/14yannick/esphome-hcpbridge) port of this project since two year it remove the need of changing source code file to make it work with other devices or sensors.

**Comment by stefasa** [comment]
Hi @Gifford47 and community,

I came across this discussion and I'm very interested in the HCPBridge Rev. 2.2 board.
I have a Hörmann SupraMatic E Serie 4 and want to integrate it into Home Assistant using ESPHome.

A few questions:

1. Is the board still available? Can I still order via the process mentioned above?

2. I'd also like to connect the following sensors directly to the board:
- DHT22 (temperature & humidity)
- HC-SR501 PIR motion sensor
- HC-SR04P ultrasonic sensor

Has anyone already designed or printed an enclosure that fits the HCPBridge PCB and has space for these sensors?
Any STL links or models would be much appreciated!

Thanks!

BR,
stefasa

**Reply by Gifford47** [reply]
Hi!
Please write to Gifford47@protonmail.com

Best regards

**Comment by jamescorday** [comment]
Hi,

I would be extremely interested in the HCPMini if these are still available, I'll be getting the Supramatic for my Hörmann sectional garage door.

Thanks

**Reply by Gifford47** [reply]
please see https://github.com/Gifford47/HCPBridgeMqtt/discussions/83#discussion-6917946 -> how to order

**Comment by harendal** [comment]
It’s possible to send HCPMini to Poland?
